app/src/main/java/org/mozilla/fenix/wallpapers/WallpaperFileManager.kt

```diff
@@ -49,17 +50,126 @@ class WallpaperFileManager(
         }
 
     /**
-     * Remove all wallpapers that are not the [currentWallpaper] or in [availableWallpapers].
+     * Get the list of migrated legacy wallpapers.
+     *
+     * @param remoteAvailableWallpapers List of wallpapers available remote.
+     * @param localAvailableWallpapers List of wallpapers available locally.
+     * @return List of migrated wallpapers.
      */
-    suspend fun clean(currentWallpaper: Wallpaper, availableWallpapers: List<Wallpaper>) = withContext(Dispatchers.IO) {
-        scope.launch {
-            val wallpapersToKeep = (listOf(currentWallpaper) + availableWallpapers).map { it.name }
-            wallpapersDirectory.listFiles()?.forEach { file ->
-                if (file.isDirectory && !wallpapersToKeep.contains(file.name)) {
-                    file.deleteRecursively()
```

Comment:
We should check with product before removing this `clean` method. It is intended to remove promotional wallpaper files once those promotions expire. During the Turning Red promotion, there was quite a bit of focus on making sure users would not be able to access Turning Red assets that they had not intended to keep. I would be concerned that we might have similar requirements from promotional partners in the future.